In [1]:
import pandas as pd
import numpy as np
import mglearn
X, y=mglearn.datasets.load_extended_boston()

C:\Users\taeeu\anaconda3\lib\site-packages\sklearn\utils\deprecation.py:87: FutureWarning: Function load_boston is deprecated; `load_boston` is deprecated in 1.0 and will be removed in 1.2.

    The Boston housing prices dataset has an ethical problem. You can refer to
    the documentation of this function for further details.

    The scikit-learn maintainers therefore strongly discourage the use of this
    dataset unless the purpose of the code is to study and educate about
    ethical issues in data science and machine learning.

    In this special case, you can fetch the dataset from the original
    source::

        import pandas as pd
        import numpy as np

        data_url = "http://lib.stat.cmu.edu/datasets/boston"
        raw_df = pd.read_csv(data_url, sep="\s+", skiprows=22, header=None)
        data = np.hstack([raw_df.values[::2, :], raw_df.values[1::2, :2]])
        target = raw_df.values[1::2, 2]

    Alternative datasets include the California housing dataset (i

In [4]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test=train_test_split(X, y, test_size=0.2, random_state=0)

from sklearn.model_selection import GridSearchCV
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
from sklearn.metrics import mean_squared_error

lr=LinearRegression()
lr.fit(X_train, y_train)

print(lr.score(X_test, y_test))
print(np.sqrt(mean_squared_error(y_test, lr.predict(X_test))))

0.615885858407892
5.592657237078517


선형회귀 모형 : 결정계수는 61%, rmse값은 5.59

In [7]:
#warinng 무시
import warnings
warnings.filterwarnings('ignore')

In [9]:
alpha=np.arange(0, 1.1, 0.1)

ridge=Ridge(normalize=True)
param_grid={'alpha':alpha}
ridge_model=GridSearchCV(ridge, param_grid)
ridge_model.fit(X_train, y_train)

print(ridge_model.best_params_)

{'alpha': 0.1}


In [10]:
ridge=Ridge(normalize=True, alpha=0.1)
ridge.fit(X_train, y_train)

print(ridge.score(X_test, y_test))
print(np.sqrt(mean_squared_error(y_test, ridge.predict(X_test))))

0.7356090234954484
4.6399293657141945


릿지회귀 : 정확도-0.73, rmse-4.63

In [11]:
lasso=Lasso(normalize=True)
param_grid={'alpha':alpha}
lasso_model=GridSearchCV(lasso, param_grid)
lasso_model.fit(X_train, y_train)
print(lasso_model.best_params_)

{'alpha': 0.0}


In [12]:
lasso=Lasso(normalize=True, alpha=0)
lasso.fit(X_train, y_train)

print(lasso.score(X_test, y_test))
print(np.sqrt(mean_squared_error(y_test, lasso.predict(X_test))))

0.6901880385280231
5.022698918446969


랏소회귀 : 정확도-69%, rmse=5.02

릿지회귀의 정확도가 더 좋은 것을 확인할 수 있음